app/src/main/java/de/danoeh/antennapod/ui/screen/playback/SleepTimerDialog.java

```diff
@@ -40,8 +51,65 @@ public class SleepTimerDialog extends DialogFragment {
     private LinearLayout timeSetup;
     private LinearLayout timeDisplay;
     private TextView time;
+    private Spinner sleepTimerType;
     private CheckBox chAutoEnable;
 
+    Button extendSleepFiveMinutesButton;
+    Button extendSleepTenMinutesButton;
+    Button extendSleepTwentyMinutesButton;
+
+    static class SleepEntryConfig {
+        public final int displayValue;
+        public final int configuredValue;
+
+        public SleepEntryConfig(int displayValue, int configuredValue) {
+            this.displayValue = displayValue;
+            this.configuredValue = configuredValue;
+        }
+    }
+
+    static class SleepTimeConfig {
+        public final int buttonTextResourceId;
+        public final int displayTypeTextId;
+        public final List<SleepEntryConfig> sleepEntries = new ArrayList<>(3);
+
+        public SleepTimeConfig(
+                int buttonTextResourceId,
+                int displayTypeTextId,
+                SleepEntryConfig first,
+                SleepEntryConfig second, SleepEntryConfig third) {
+            this.buttonTextResourceId = buttonTextResourceId;
+            this.displayTypeTextId = displayTypeTextId;
+            sleepEntries.add(0, first);
+            sleepEntries.add(1, second);
+            sleepEntries.add(2, third);
+        }
+    }
+
+    private static <K, V> LinkedHashMap<K, V> linkedMapOf(Collection<Map.Entry<? extends K, ? extends V>> entries) {
+        LinkedHashMap<K, V> map = new LinkedHashMap<>();
+        for (Map.Entry<? extends K, ? extends V> entry : entries) {
+            map.put(entry.getKey(), entry.getValue());
+        }
+        return map;
+    }
+
+    LinkedHashMap<SleepTimerType, SleepTimeConfig> allSleepConfig = linkedMapOf(List.of(
```

Comment:
This feels more generic than it needs to be. I don't think there will be additional types of sleep timers anytime soon. Making the code less generic and more "manual" would make it easier to understand for future developers.